### PR TITLE
Feature/122 enhance raw transaction view

### LINKED ISSUE #122: Feature: enhance view on display the raw transaction
> As **User** I'd like to know the payload I'm authorizing before starting the signing process and be provided with more information.

### ⚠️ what do we need?

- [ ] UI/UX 
- [ ] Copywriting

### Acceptance criteria:

- [x] Before user being presented to `sign` the tx, user can have a preview of the raw payload
- [x] when click `sign`, user will be presented with the `QR` code, what the app currently has
- [x] the raw payload is collapsed first, only expend with user interaction
- [x] user get an estimate on fees (see screenshot)

### Detailed Description (if necessary)
- _[...]_

### implementation details (if necessary)
* Collapse the detail payload first
* Enhance the current HOC to be able to pass additional `tx` payload to present the payload
* Possibly need a mapping for `module.methodName` <=> brief description

Screenshot 2021-06-11 at 02 33 29

### PR DESCRIPTION
**Pull request by ahoseinian**
### Description
- Refactor Transaction context to use better state management
- Add Preview page to transactions to see details and fees applied

part of #122 

https://user-images.githubusercontent.com/1168983/121893458-4416f400-cd1e-11eb-886e-9b4dcc23910b.MP4

### CODE REVIEW

**Comment by ahoseinian** [comment]
@franjs @geastwood can I get a review here?

File: src/context/TxContext/PreviewStep.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import {ScrollView, StyleSheet, TouchableOpacity, View} from 'react-native';
+import globalStyles, {standardPadding} from 'src/styles';
+import {HashBlock} from 'presentational/HashBlock';
+import Padder from 'presentational/Padder';
+
+type PropTypes = {
+  transactionTitle: string;
+  transactionInfo: string;
+  payload: SignerPayloadJSON;
+  params: unknown;
```
Comment (silva-fj) [inline src/context/TxContext/PreviewStep.tsx]: should `params` be an array?

File: src/context/TxContext/index.tsx [hunk 2]
```diff
@@ -86,8 +62,12 @@ function TxContextProvider({children}: PropTypes) {
         nonce: -1,
         tip: BN_ZERO,
         signer: new QrSigner((payload) => {
-          setSignerPayload(payload);
-          setStep('payload');
+          transaction.paymentInfo(address).then((info) => {
+            dispatch({
+              type: 'PREVIEW',
+              payload: {payload, params, title, description, partialFee: info.partialFee.toNumber()},
```
Comment (silva-fj) [inline src/context/TxContext/index.tsx]: how about `payload: {txPayload, ....` ?

File: src/context/TxContext/index.tsx [hunk 3]
```diff
@@ -102,63 +82,76 @@ function TxContextProvider({children}: PropTypes) {
...
-                  return documentation.join(' ').trim();
-                } else {
-                  // Other, CannotLookup, BadOrigin, no extra info
-                  return error.toString();
-                }
-              },
-            );
+              } = eventRecord;
+
+              if ((error as any).isModule) {
```
Comment (silva-fj) [inline src/context/TxContext/index.tsx]: why `any`? There should be a type for this error

File: src/context/TxContext/index.tsx [hunk 4]
```diff
@@ -102,63 +82,76 @@ function TxContextProvider({children}: PropTypes) {
...
+                // for module errors, we have the section indexed, lookup
+                const decoded = api.registry.findMetaError((error as any).asModule);
```
Comment (silva-fj) [inline src/context/TxContext/index.tsx]: same

File: src/presentational/IdentityInfoForm.tsx [hunk 5]
```diff
@@ -3,32 +3,26 @@ import {View, StyleSheet, ScrollView} from 'react-native';
...
+  legal: {raw: string} | {none: null};
+  email: {raw: string} | {none: null};
+  riot: {raw: string} | {none: null};
+  twitter: {raw: string} | {none: null};
+  web: {raw: string} | {none: null};
+};
 
 type IdentityInfoFormProps = {
-  onSubmit: (identityInfo: IdentityInfo) => void;
+  onSubmit: (identityInfo: SubmitIdentityPayload) => void;
```
Comment (silva-fj) [inline src/presentational/IdentityInfoForm.tsx]: NIT: `IdentityInfo` or `IdentityPayload` would cleaner IMO

File: src/presentational/TxPayloadQr.tsx [hunk 6]
```diff
@@ -1,39 +1,40 @@
...
-    wrapper.toU8a(),
-    payload.genesisHash,
-  );
 
   useEffect(() => {
+    if (!api) {
+      return;
+    }
+    // limit size of the transaction
+    const isQrHashed = payload.method.length > 5000;
```
Comment (silva-fj) [inline src/presentational/TxPayloadQr.tsx]: Can you move the number to a constant and remove the comment?

Like:
`const TX_SIZE_LIMIT = 5000;`

File: src/screen/MyIdentityScreen/RequestJudgement.tsx [hunk 7]
```diff
@@ -40,7 +40,14 @@ function RequestJudgement({display, address, detail}: PropTypes) {
...
     async (index: number, fee: BN) => {
       if (api) {
-        await start(api, address, 'identity.requestJudgement', [index, fee]);
+        await start({
+          api,
+          address,
+          txMethod: 'identity.requestJudgement',
+          params: [index, fee],
+          title: `Sending transaction identity.requestJudgement`,
+          description: 'TODO',
```
Comment (silva-fj) [inline src/screen/MyIdentityScreen/RequestJudgement.tsx]: what's TODO?

File: src/context/TxContext/PreviewStep.tsx (hunk 1 above)
Comment (ahoseinian) [reply src/context/TxContext/PreviewStep.tsx]: yes, I have changed it 👍

File: src/context/TxContext/index.tsx (hunk 3 above)
Comment (ahoseinian) [reply src/context/TxContext/index.tsx]: yes you are right, I found an example and did the same 
https://github.com/polkadot-js/apps/blob/d1389052c43c24585ce87b17da8a82dc6e8294e9/packages/react-components/src/Status/Queue.tsx#L88

File: src/screen/MyIdentityScreen/RequestJudgement.tsx (hunk 7 above)
Comment (ahoseinian) [reply src/screen/MyIdentityScreen/RequestJudgement.tsx]: forgot to fix this, it's now `Request a judgement from a registrar.`

**Review by silva-fj — APPROVED** [review]
Good improvements 👍🏼

**Pull request merged**